### ISSUE TITLE
False positive: parameter must be non-null in inner class constructor 

### ISSUE THREAD

**Issue by ahubold** [spotbugs/spotbugs#772]
Spotbugs 3.1.6 shows an issue for rule findbugs:NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE that an inner class constructor parameter must be non-null, even if the parameter is already annotated as NonNull. Code example:

```
package com.example;

import edu.umd.cs.findbugs.annotations.NonNull;
import edu.umd.cs.findbugs.annotations.Nullable;

public class NonNullParameterOfInnerClassConstructor {

  private class Inner {
    private final String a;
    private final Object b;

    Inner(@NonNull String a, @Nullable Object b) {
      this.a = a.toLowerCase(); // <--- false positive: a must be non-null but is marked as nullable
      this.b = b;
    }
  }
}

**Comment by KengoTODA** [comment]
@ahubold Thanks for your nice report and reproduction! I've fixed it, we'll release patched version in the next January.
